assets/src/models/ladderVehicle.ts

```diff
@@ -1,11 +1,27 @@
 import { LadderDirection, TimepointStatusYFunc } from "../components/ladder"
-import { DirectionId, Vehicle } from "../skate"
+import {
+  DirectionId,
+  Vehicle,
+  VehicleAdherenceStatus,
+  VehicleId,
+  ViaVariant,
+} from "../skate"
+import {
+  isUnassignedBySwiftly as vehicleIsUnassignedBySwiftly,
+  status,
+} from "./vehicleStatus"
 
 export interface LadderVehicle {
-  vehicle: Vehicle
+  vehicleId: VehicleId
+  label: string
+  viaVariant: ViaVariant | null
+  status: VehicleAdherenceStatus
+  isUnassignedBySwiftly: boolean
   x: number
   y: number
   vehicleDirection: VehicleDirection
+  scheduledY?: number
+  scheduledVehicleDirection?: VehicleDirection
```

Comment:
The whole point of these was to show intermediate stages of data and make it clear that certain functions only require some subsets of properties. `LadderVehicle` is the only thing that matters publicly though, so I think it's valuable to keep everything explicitly defined there, rather than having to try to decipher the results of a chain of `extends`.